js-exercises/array-utils/arrayUtils.js

```diff
@@ -0,0 +1,57 @@
+function forEach(array, callbackFn) {
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
+      callbackFn(array[i], i, array)  
+    }
+  }
+
+  return;
+}
+
+function map(array, callbackFn) {
+  let resultingArray = [];
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
+      resultingArray.push(callbackFn(array[i], i, array));
+    }
+  }
+
+  return resultingArray;
+}
+
+function filter(array, callbackFn) {
+  let resultingArray = [];
+  if(!array.length) throw new TypeError('Passed argument is not an array');
+
+  if(typeof callbackFn !== "function") throw new TypeError('Requires a callback function to execute');
+
+  let arrayLength = array.length;
+  
+  for(let i = 0; i < arrayLength; i++) {
+    if(array[i]) {
```

Comment:
Due to this check `if(array[i])` different behaviour than Array.prototype.filter:
`filter([1, null, 3, 4], (element, index) => element < 4)`
> (2) [1, 3]

`[1, null, 3, 4].filter((el, i) => el < 4)`
> (3) [1, null, 3]

Reply:
Thanks, Yash for spotting the errors out! What I am going to do is, I am going to rewrite all of the above implementations the way Arfat showed us in the previous lecture! By following the ES spec steps, all the cases will get handled.